api/v1alpha1/backend_types.go

```diff
@@ -0,0 +1,153 @@
+	KindBackend = "Backend"
+)
+
+// +kubebuilder:validation:Enum=gateway.envoyproxy.io/h2c;gateway.envoyproxy.io/ws;gateway.envoyproxy.io/wss
+// +notImplementedHide
+type AppProtocolType string
+
+const (
+	// AppProtocolTypeH2C defines the HTTP/2 application protocol.
+	AppProtocolTypeH2C AppProtocolType = "gateway.envoyproxy.io/h2c"
+	// AppProtocolTypeWS defines the WebSocket over HTTP protocol.
+	AppProtocolTypeWS AppProtocolType = "gateway.envoyproxy.io/ws"
+	// AppProtocolTypeWSS defines the WebSocket over HTTPS protocol.
+	AppProtocolTypeWSS AppProtocolType = "gateway.envoyproxy.io/wss"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=envoy-gateway,shortName=be
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[?(@.type=="Accepted")].reason`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +notImplementedHide
+//
+// Backend allows the user to configure the behavior of the connection
+// between the Envoy Proxy listener and the backend service.
+type Backend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec defines the desired state of Backend.
+	Spec BackendSpec `json:"spec"`
+
+	// status defines the current status of Backend.
+	Status BackendStatus `json:"status,omitempty"`
+}
+
+// BackendAddress describes are backend address, which is can be either a TCP/UDP socket or a Unix Domain Socket
+// corresponding to Envoy's Host: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/address.proto#config-core-v3-address
+//
+// +kubebuilder:validation:XValidation:rule="(has(self.fqdn) || has(self.ip) || has(self.unix))",message="one of fqdn, ip or unix must be specified"
+// +kubebuilder:validation:XValidation:rule="((has(self.fqdn) && !(has(self.ip) || has(self.unix))) || (has(self.ip) && !(has(self.fqdn) || has(self.unix))) || (has(self.unix) && !(has(self.ip) || has(self.fqdn))))",message="only one of fqdn, ip or unix can be specified"
+// +notImplementedHide
+type BackendAddress struct {
+	// FQDN defines a FQDN address
+	//
+	// +optional
+	FQDN *FQDNAddress `json:"fqdn,omitempty"`
+
+	// IP defines a IPv4 address
+	//
+	// +optional
+	IP *IPAddress `json:"ip,omitempty"`
+
+	// Unix defines the unix domain socket path
+	//
+	// +optional
+	Unix *UnixSocket `json:"unix,omitempty"`
+}
+
+// IPAddress describes TCP/UDP socket address, corresponding to Envoy's Socket Address
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/address.proto#config-core-v3-socketaddress
+// +notImplementedHide
+type IPAddress struct {
+	// Host defines to the IPv4 address of the backend service.
+	//
+	// +kubebuilder:validation:MinLength=7
+	// +kubebuilder:validation:MaxLength=15
+	// +kubebuilder:validation:Pattern=`^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$`
+	Host string `json:"host"`
```

Comment:
`Host` or `Address` ?

Reply:
With endpoints as a top-level, we can have address here, which I feel is more accurate. Updated it now,